File: Back-end/biblioteca/src/main/java/com/managerServices/biblioteca/exception/HttpReturnExceptionHandler.java
package com.managerServices.biblioteca.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.context.request.WebRequest;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

import com.managerServices.biblioteca.exception.response.ErrorResponse;

@ControllerAdvice
@RestController
public class HttpReturnExceptionHandler extends ResponseEntityExceptionHandler {
	
	@ExceptionHandler(HttpReturnException.class)
	public final ResponseEntity<ErrorResponse> handleProviderError(HttpReturnException e, WebRequest request) {
		
		ErrorResponse res = new ErrorResponse();
		res.setMessage(e.getMessage());
		res.setStatus(e.getStatus());
		res.setStatus_code(e.getStatus_code());
		
		return new ResponseEntity<ErrorResponse>(res, e.getStatus());

	}
	

	
}

File: Back-end/biblioteca_kotlin/library/src/main/kotlin/com/biblioteca/library/exception/HttpReturnExceptionHandler.kt
package com.biblioteca.library.exception

import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.http.ResponseEntity
import com.biblioteca.library.exception.response.ErrorResponse
import org.springframework.web.context.request.WebRequest

@ControllerAdvice
@RestController
class HttpReturnExceptionHandler : ResponseEntityExceptionHandler() {
	
	@ExceptionHandler(HttpReturnException::class)
	final fun handleProviderError(e: HttpReturnException, request: WebRequest) : ResponseEntity<ErrorResponse>{
		
		var res = ErrorResponse();
		res.message = e.message
		res.status =  e.status
		res.status_code = e.status_code
		
		return ResponseEntity<ErrorResponse>(res, e.status)
		
	}
}